### DISCUSSION TITLE
How to use makefile to copy the library and inc file of dependencies after built?

### DISCUSSION THREAD

**Discussion by LuoSYLuo** [eclipse-cdt/cdt#400] [Q&A]
I am the new developer. one thing that I want to do is copy project's library file and header file of the library dependencies:
And I need to complete this build by makefile(GNU Makefile)

ex:
-ProjectA
--inc
 --ProjectA_API.h
 --ProjectA_Type.h
--lib
  --Release
     --ProjectA.a

-Main
--bin
--src
Makefile

How can I complete this build project task by makfile to copy the library dependencies?
does it use many make variables(ex: PROJECRT_NAME, PROJECT_HOME) or makefile.include to write the makefile?

Becasue I can use post-build and write the batch file to copy all project's library file and header file, it will copy the library not included in this project.

ex:
for /R C:\usr\xtensa\AAA %%f in (*.a) do copy "%%f" C:\usr\xtensa\AAA\

**Comment by jonahgraham** [comment]
You can add a `makefile.targets` file in the root of your project with any makefile targets you want. For example, if you want something copied before the final link and your project is called `simple` adding a target like this to the `makefile.targets` would probably work:

```makefile
simple: ../path/to/file
	cp $< .
```
